queries/teal/locals.scm

```diff
@@ -0,0 +1,20 @@
+
+(var_declaration
+   (var (identifier)) @definition.var)
+
+(var_assignment
+   (var (identifier)) @definition.associated @definition.var)
+
+(arg_name) @definition.parameter
+
+(functiondef) @scope
+(function_statement (function_name) @definition.function) @scope
```

Comment:
You need to set the scope for functions

```
(#set! definition.function.scope "parent")
```

Reply:
I'm not super familiar with the predicates/query syntax yet, that would go inside the query nodes?
```
(function_statement
  (function_name) @definition.function
  (#set! definition.function.scope "parent")) @scope
```

Reply:
It would be something like

```
((function_statement (function_name) @definition.function) @scope
 (#set! definition.function.scope "parent"))
```